plugins/Course/rmpScraper.py

```diff
@@ -0,0 +1,80 @@
+from bs4 import BeautifulSoup
+from urllib.request import urlopen as uReq
+
+# These are the parts that go into a RMP search for a MUN prof (minus the name)
+urlParts = [
+    "https://www.ratemyprofessors.com/search/teachers?query=",
+    "&sid=U2Nob29sLTE0NDE=",
+]
+
+# Get the completed URL for the RMP search
+def getRMPURL(separatedProfName):
+
+    finalUrl = urlParts[0]
+
+    # Append every word from the name to the URL with "%20" in between them
+    finalUrl += "%20".join(separatedProfName)
+
+    finalUrl += urlParts[1]
+    return finalUrl
+
+
+# Get the soup from a URL
+def getSoupFromURL(url):
```

Comment:
could you implement some basic caching here? currently anytime a user runs `!course`, this will fetch rmp, would be nice to not get rate limited / banned :sweat_smile: 

theres some example cache code within the whitelist plugin for caching requests to the mojang api using mongodb collections that expire documents after a set amount of time

Reply:
Done